### PR TITLE
Default log file for nrunner based tests

### PR DESCRIPTION
**Pull request by richtja**
This will save every nrunner messages from stdout and stderr to the `debug.log` with the proper tag on the beginning of each line, and it introduces the `debug.log` file as Default log file for nrunner.

Reference: #4804
Signed-off-by: Jan Richter <jarichte@redhat.com>

### CODE REVIEW

File: avocado/core/messages.py [hunk 1]
```diff
@@ -157,6 +159,51 @@ def handle(self, message, task, job):
...
     """Base interface for resolving running messages."""
 
+    _tag = b''
+
+    def __init__(self):
+        self.line_buffer = b''
+
+    def _get_tagged_line(self, data, encoding=None):
+        """
+        It will add the tag in to each line of data as a prefix. If the data
```
Comment (beraldoleal) [inline avocado/core/messages.py:172]: PEP257, please.

File: avocado/core/messages.py [hunk 2]
```diff
@@ -281,7 +332,10 @@ class StderrMessageHandler(BaseRunningMessageHandler):
              'time': 18405.55351474}
     """
 
+    _tag = b'[stdout] '
```
Comment (beraldoleal) [inline avocado/core/messages.py:338]: IIUC, s/stdout/stderr/g since this is the StderrMessageHandler?

File: avocado/core/messages.py [hunk 3]
```diff
@@ -157,6 +159,51 @@ def handle(self, message, task, job):
...
+        The default log file is based on `DEFAULT_LOG_FILE` variable and every
+        line of log will be saved with prefix based on `_tag` variable
+
+        :param message: message from runner
+        :type message: dict
+        :param task: runtime_task which message is related to
+        :type task: :class:`avocado.core.nrunner.Task`
+        """
+        encoding = message.get('encoding', None)
+        data = self._get_tagged_line(message['log'], encoding)
```
Comment (beraldoleal) [inline avocado/core/messages.py:206]: This method name is indicating that we are going to "get something", but in fact this method is also setting some object property (line 179, 187). I would suggest to refactor this.

Comment (beraldoleal) [inline avocado/core/messages.py:206]: Also, `log` and `encoding` are both part of `message`. So maybe just pass the `message` to the method.

File: avocado/core/messages.py [hunk 4]
```diff
@@ -157,6 +159,51 @@ def handle(self, message, task, job):
...
+
+        The default log file is based on `DEFAULT_LOG_FILE` variable and every
+        line of log will be saved with prefix based on `_tag` variable
+
+        :param message: message from runner
+        :type message: dict
+        :param task: runtime_task which message is related to
+        :type task: :class:`avocado.core.nrunner.Task`
+        """
+        encoding = message.get('encoding', None)
```
Comment (beraldoleal) [inline avocado/core/messages.py:205]: None is not needed here.

File: avocado/core/messages.py [hunk 5]
```diff
@@ -157,6 +159,51 @@ def handle(self, message, task, job):
...
+    _tag = b''
+
+    def __init__(self):
+        self.line_buffer = b''
+
+    def _get_tagged_line(self, data, encoding=None):
+        """
+        It will add the tag in to each line of data as a prefix. If the data
+        don't finish with the new line character, the last line is marked
+        as incomplete, and it is saved to the line buffer for next usage.
```
Comment (beraldoleal) [inline avocado/core/messages.py:174]: This last part that I'm talking about on a following comment. Maybe we could let each method do only one single thing and the split/check for incomplete lines could be done in a separated method.

**Review by beraldoleal — CHANGES_REQUESTED** [review]
Hi @richtja, thanks. I left a few comments for you.

File: avocado/core/messages.py (hunk 5 above)
Comment (richtja) [reply avocado/core/messages.py:174]: Ok, I will refactor this.

**Comment by richtja** [comment]
@beraldoleal Thanks for your comments, I will prepare new version.

**Comment by richtja** [comment]
Closing in favor of #4876

**Pull request closed without merging**